### PR TITLE
Enable the user to be able login and out

### PR DESCRIPTION
**Pull request by elvongray**

### CODE REVIEW

**Comment by mxfactorial** [comment]
Hardcoding 'localhost:8080' instead of referencing BASE_URL in app/constants/index.js causes 'Fetch API cannot load' error in alternative environment.

1) https://github.com/elvongray/mxfactorial/blob/386a19ffe5373df388859f91d88c8214bb736bee/app/actions/signUpActions.js#L34
2) https://github.com/elvongray/mxfactorial/blob/386a19ffe5373df388859f91d88c8214bb736bee/app/components/CreateAccountForms/sixthForm.js#L34

**Comment by elvongray** [comment]
@mxfactorial I am still working on it, not yet through.

**Comment by mxfactorial** [comment]
Added a "ready: request testing" label to the project.

**Comment by elvongray** [comment]
@mxfactorial The option to select a label is disabled for me. I have completed the feature. I also did some refactoring, but in general, the code still needs to more refactoring(trying to squeeze time for that). I have some suggestions though.
1. The UI/UX of the application needs to be improved. I am not really good at design, so I can't be much of help.
2. Also the api needs to be improved, not really helpful when trying to implement the login feature(breaks a lot and I had to do some hacks to make it work on the frontend)

**Comment by mxfactorial** [comment]
Will use comments instead of labels. Updated the delivery doc.

Please indicate whether inefficient code is increasing the difficulty of working with the UI/UX , or you're only pointing out aesthetics. If it's the former, will locate someone to clean up the UI for easy addition of other developers. If you're only pointing out the visual appeal, then yes, this client requires a cosmetic overhaul that is intended to occur after all the buttons and forms are tested to pass/receive data as expected to/from the Datastore.
The comment on the API is noted. It's the first time @bryantcj5691 has worked with the Datastore. He's likely interested in learning how to improve the API. Do you have any further details to offer for his investigation? Were your requests erroring when they should not have been? Responses failing or contained incomplete payloads? What standard behavior does your client expect that it wasn't receiving?
Thank you. Will test this request when available tomorrow.

**Comment by elvongray** [comment]
@mxfactorial The UI/UX improvement suggestion is just based on aesthetics. For the api, the response messages can still be improved, request were failing lots of times that I needed to restart the server, server breaks easily and so on. I wil try to find examples to help guide @bryantcj5691 on improving the authentication process.

**Comment by bryantcj5691** [comment]
@mxfactorial  ,  @elvongray   sorry for the late reply. @elvongray  any examples to make things better are welcome. I'll also try to find out how the response can be improved. @mxfactorial  I couldn't find time to complete the task on modifying account creation endpoint. is @kishanj918  waiting for me to complete that task? Hope to complete this soon. 

shall i add this to the project?
http://apidocjs.com/

developers may find it easy to get information about endpoints

**Comment by mxfactorial** [comment]
Kishan is occupied with other work at the moment. And no, there isn't an immediate client dependency on improving the API, but we will begin this improvement immediately by adding the apidoc package you've recommended, rigorously document the API, then identify features client developers such as @elvongray value.

**Comment by mxfactorial** [comment]
Bug 1: 
1. Complete the create account sequence to add entity to the Datastore.

Observed: Only email_address_create, password_create, and user_create properties are received in a user entity by the Datastore.

Expected: All properties defined in the create account sequence are received by the Datastore.

Bug 2:
1. Complete the create account sequence to display the homeScreen.
2. Select burgerIcon to display the mainNav.
3. Select the 'Log out' menu item to display the landingScreen.
4. Select the createButton.
5. Select the nextButton twice.
6. Select the iAcceptButton.
7. Insert a different user, password, & email address.
8. Select the Next button.

Observed: 'Unable to authenticate username' pop-up displays along with a console printing "POST http://localhost:8080/systemaccounting/users 500 (Internal Server Error) @ sixthForm.js:18".

Expected: Next screen in the create account sequence is displayed, "First Name_, Middle Name_, etc."

@elvongray, if you suspect these issues are appearing because of client-side code, fix when available. However, if you're convicting API code as source, then we will identify and remedy those issues first. Will avoid wasting your time on hacking fixes since these endpoints are intended to be publicly-accessible.

Awaiting your response.

**Comment by elvongray** [comment]
@mxfactorial For bug 1, the api currently just saves the  email_address_create, password_create, and user_create. The remaining properties are not currently handled.

@) For bug 2, it is definitely a server error. anytime such occurs, I usually restart the server.

**Comment by mxfactorial** [comment]
**Re: Bug 1**
"The remaining properties are not currently handled." - Cannot reproduce this statement with master on production: https://drive.google.com/file/d/0B9xlXsaN9dVQWEVGR2xsYW53Yms/view?usp=sharing

Here's another theory: Becoming weary of clicking though all the account creation screens, @elvongray moved the auth-related screen to the front of the sequence, thus unintentionally excluding subsequent values because the accompanying POST request was also moved :)

**Re: Bug 2**
Let's fix bug 1 first, then test bug 2 again before continuing a diagnosis.

Also, testing documentation was added to the wiki to include the public in this process.

**Comment by elvongray** [comment]
@mxfactorial You are right, I can see my error. Will resolve it.

**Comment by elvongray** [comment]
@mxfactorial I have resolved bug 1

**Comment by mxfactorial** [comment]
Please eliminate conflicts. Next in line for merging.

File: app/client.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const store = configureStore();
 ReactDOM.render((
     <Provider store={store}>
         <Router history={ hashHistory }>
-            { getRoutes() }
+            { getRoutes(store) }
```
Comment (hotay) [inline app/client.js:23]: Why did you pass store here? it's already available in react context

**Comment by mxfactorial** [comment]
This feature is now the highest priority. Please see https://github.com/systemaccounting/mxfactorial/issues/24#issuecomment-230043930

**Comment by mxfactorial** [comment]
Replaced by https://github.com/systemaccounting/mxfactorial/pull/50.

**Pull request closed without merging**